base/dcp_client.go

```diff
@@ -443,6 +446,12 @@ func (dc *DCPClient) openStream(vbID uint16, maxRetries uint32) error {
 				return fmt.Errorf("%s, failOnRollback requested", openStreamErr)
 			}
 			InfofCtx(logCtx, KeyDCP, "Open stream for vbID %d failed due to rollback or range error, will roll back metadata and retry: %v", vbID, openStreamErr)
+			if dc.attachmentCompactionClient {
```

Comment:
I think it's conceptually easier to think about this is we make this `failOnRollback` and then make a public method to purge metadata - and handle that purge in the calling code (attachment compaction).

If we need to purge while the client is still alive, I'd create an option where you failOnRollback that does failOnRollback and purge metadata, but it seems pretty niche.

You could type the RollbackError to make it easier to find from attachment compaction if that makes it easier.